src/main/kotlin/main.kt

```diff
@@ -140,51 +78,28 @@ class AudioGraph(val ui: Ui) {
         // Load the desired audio clip (GET it and decode the raw data buffer):
         // TODO: audio buffers are reusable, load+decode them only once
 
-        XMLHttpRequest().apply {
-
-            open("GET", clip, true)
-            responseType = XMLHttpRequestResponseType.ARRAYBUFFER
-
-            onerror = {
-                onRejected(Exception("$clip request error: $it"))
-            }
+        val encodedData = window.fetch(clip).await().arrayBuffer().await()
+        // once we got the array buffer with the clip data
+        // we need to decode that, and set the result into the
+        // BufferSourceNode:
+        source.buffer = ctx.decodeAudioData(encodedData).await()
+        source.loop = true
 
-            onload = {
-
-                val encodedData = response as? ArrayBuffer
-
-                if (encodedData == null) {
-                    println("Null audio data from response!")
-                } else {
-
-                    // once we got the array buffer with the clip data
-                    // we need to decode that, and set the result into the
-                    // BufferSourceNode:
-
-                    ctx.decodeAudioData(encodedData).then { decoded ->
-                        source.buffer = decoded
-                        source.loop = true
-                        onResolved(source)
-                    }
-                }
-            }
-
-            // Fire the request:
-            send()
-        }
+        return source
     }
 
     private fun stop() {
-        context?.let {
-            sources?.forEach { it.stop() }
-            sources = null
-            it.close()
-            context = null
-        }
+        sources?.forEach { it.stop() }
+        sources = null
+
+        context?.close()
+        context = null
     }
 }
 
 fun main() {
-    val ui = Ui(document)
-    AudioGraph(ui)
+    window.onload = {
```

Comment:
@gildor what's the difference with just running the code in main?

Reply:
It allows include script to header, so program doesn't depends on script include order